### PR TITLE
win: try to find VS2017 tooling

### PR DESCRIPTION
**Pull request by refack**
Using the MS recommended way of looking in the registry
Work with all flavors of VS2017, even the minimal "C++ Build Tools"
(obviously `npm run test` passes)

### CODE REVIEW

**Comment by refack** [comment]
@indutny review?

**Comment by refack** [comment]
@indutny BTW don't give up on this project!
I hate the `gyp` dependency. It's virtually an abandoned project, it drags in python, and it just makes the build environment too brittle (more so for "native modules").

File: lib/gyp/platform/win.js [hunk 1]
```diff
@@ -411,65 +411,95 @@ win.getOSBits = function getOSBits() {
...
-                   envBlock);
+  if (!outDir) {
+    return lines;
+  }
 
+  const envBlock = formatEnvBlock(lines);
+  const envFile = 'environment.' + arch;
+  fs.writeFileSync(path.join(outDir, envFile), envBlock);
   return envFile;
+
```
Comment (indutny) [inline lib/gyp/platform/win.js]: Superfluous change

File: test/platform-win-test.js [hunk 2]
```diff
@@ -13,38 +13,38 @@ describe('gyp.platform.win', () => {
 
     it('should remove prefix `-l`', () => {
       assert.deepEqual(
-        win.adjustLibraries([ '-llib1.lib', 'lib2.lib' ]),
-        [ 'lib1.lib', 'lib2.lib' ]);
+        win.adjustLibraries(['-llib1.lib', 'lib2.lib']),
```
Comment (indutny) [inline test/platform-win-test.js]: Any reason to change this?

**Comment by refack** [comment]
So the "Setup.Configuration.exe" tools comes from Microsoft's [recomanded way to find visual studio components](https://blogs.msdn.microsoft.com/heaths/2016/09/15/changes-to-visual-studio-15-setup/) actualy [this repos](https://github.com/Microsoft/vs-setup-samples), self compiled. It's a tiny tools that just calls a COM server, and obviously only works when VS2017 is installed but fails gracefully.
I did a similar PR at nodejs/node-gyp/#1103

P.S. White space changes were reverted.

**Comment by indutny** [comment]
@refack I have very mixed feelings about distributing `.exe` file. Is there any way to port the functionality to plain JS?

**Comment by refack** [comment]
I totally get what you're saying, I'll dig in more...

**Comment by indutny** [comment]
Thank you! I appreciate your effort!

**Comment by indutny** [comment]
Given that this indeed speaks to COM, I'm not sure if there is any easier way to do it without having either `.exe` file or custom binding... That `.exe` file is distributed under MIT license as far as I can tell, perhaps we should go with it.

What do you think?

**Comment by refack** [comment]
There is a way to talk to COM via `powershell` and even with JScript via `wscript` (Used to practice the dark art of JScript in the early 2000's). Give me the rest of the day.

**Comment by indutny** [comment]
Sure! Thanks!

**Comment by refack** [comment]
Cracked it! 🥇

**Comment by refack** [comment]
made obsolete by #33

**Pull request closed without merging**